commit 4aa3b44e10afc89f015d662c10f251ea05629f8b
Author: Alexander K <ak@natsys-lab.com>
Date:   Mon Aug 7 16:24:08 2017 +0300

    Small cleanups:
    1. Allocate at least 32MB for TempestaDB to make Tempesta FW run;
    2. Update comment about skb allocation in compound pages.

diff --git a/Documentation/kernel-parameters.txt b/Documentation/kernel-parameters.txt
index b13d0493..fcd8903a 100644
--- a/Documentation/kernel-parameters.txt
+++ b/Documentation/kernel-parameters.txt
@@ -4054,7 +4054,8 @@ bytes respectively. Such letter suffixes can also be entirely omitted.
 	tempesta_dbmem=	[KNL]
 			Order of 2MB memory blocks reserved on each NUMA node
 			for Tempesta database. Huge pages are used if
-			possible.
+			possible. Minimum value to start Tempesta is 4 (32MB).
+			Default is 8, i.e. 512MB is reserved.
 
 	test_suspend=	[SUSPEND][,N]
 			Specify "mem" (for Suspend-to-RAM) or "standby" (for
diff --git a/mm/tempesta_mm.c b/mm/tempesta_mm.c
index a131d481..8c976f0e 100644
--- a/mm/tempesta_mm.c
+++ b/mm/tempesta_mm.c
@@ -26,7 +26,7 @@
 #include "internal.h"
 
 #define MAX_PGORDER		16	/* 128GB per one table */
-#define MIN_PGORDER		0	/* 2MB - one extent */
+#define MIN_PGORDER		4	/* 32MB */
 #define DEFAULT_PGORDER		8	/* 512MB */
 /* Modern processors support up to 1.5TB of RAM, be ready for 2TB. */
 #define GREEDY_ARNUM		(1024 * 1024 + 1)
diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index d5b7bcf9..6d564e10 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -283,9 +283,9 @@ do {									\
 	/*
 	 * Don't try to split compound page.
 	 * TODO compound pages can be split as __alloc_page_frag() does it
-	 * using fragment size in page reference counter. However, it seems
-	 * the workflow is quite rare, in fact we've never seen large @size
-	 * in calls of the function.
+	 * using fragment size in page reference counter. Large messages
+	 * (e.g. large HTML pages returned by a backend server) go this way
+	 * and allocate compound pages.
 	 */
 	if (po)
 		return ptr;
